src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -421,19 +427,44 @@ async def build_pex(
     requirements_file_digest = EMPTY_DIGEST
     if request.requirements.file_path:
         argv.extend(["--requirement", request.requirements.file_path])
+
+        globs = PathGlobs(
+            [request.requirements.file_path],
+            glob_match_error_behavior=GlobMatchErrorBehavior.error,
+            description_of_origin=request.requirements.file_path_description_of_origin,
+        )
+
+        # use contents to invalidate here
+        requirements_file_digest_contents = await Get(
+            DigestContents,
+            PathGlobs,
+            globs,
+        )
+        metadata = read_lockfile_metadata(requirements_file_digest_contents[0].content)
+        if metadata.invalidation_digest != request.requirements.lockfile_hex_digest:
+            if python_setup.fail_on_invalid_lockfile:
+                raise ValueError("Invalid lockfile provided. [TODO: Improve message]")
+            else:
+                logger.warning("%s", "Invalid lockfile provided. [TODO: Improve message]")
```

Comment:
Did you mean to address this here, or it will be a followup?

Reply:
It wasn't clear to me how to provide a really meaningful error here. I'll discuss this with you out-of-band

Reply:
Sg! Because this is experimental, it's fine to land in this state imo.

Reply:
Great. Let's follow up in a separate PR.